spv/sync.go

```diff
@@ -1490,7 +1526,8 @@ func (s *Syncer) getHeaders(ctx context.Context, rp *p2p.RemotePeer) error {
 		bestChain, err := s.wallet.EvaluateBestChain(ctx, &s.sidechains)
 		if err != nil {
 			s.sidechainMu.Unlock()
-			return err
+			rp.Disconnect(err)
+			continue nextbatch
 		}
 		if len(bestChain) == 0 {
 			s.sidechainMu.Unlock()
```

Comment:
This concerns me slightly.  It looks like if a peer does present an initial height higher than the current wallet, and is able to provide these headers, but the wallet's current main chain tip is the best tip (perhaps due to the other peer's chain performing less total work) we might just keep looping and trying the same peer over and over.

Reply:
The peer picker for get headers is 

`rp.InitialHeight() > tipHeight && rp.LastHeight() < rp.InitialHeight()`

which means, if they provided the headers, `LastHeight()` will be updated, and `LastHeight() >= InitialHeight()` (because otherwise, we would've disconnected due to the protocol violation), thus that peer won't be selected again to fetch headers from.

Reply:
ah, perfect